pkg/descheduler/evictions/evictions.go

```diff
@@ -187,6 +152,66 @@ func evictPod(ctx context.Context, client clientset.Interface, pod *v1.Pod, poli
 	return err
 }
 
+type constraint func(pod *v1.Pod) error
+
+type evictable struct {
+	constraints []constraint
+}
+
+func (pe *PodEvictor) Evictable() *evictable {
+	ev := &evictable{}
+	if !pe.evictLocalStoragePods {
```

Comment:
that's not a bad idea, but have we seen any requests for this functionality yet? If it currently works for everyone as a cluster-wide option it might overcomplicate the api, and we need to consider that it would be a change from current behavior

Reply:
yeah, we may consider this change when we really need it.